### PR TITLE
CP-31555 Use bisect_ppx to generate coveralls report

### PR DESCRIPTION
**Pull request by lippirk**
The [report](https://coveralls.io/github/xapi-project/) appears to be uploaded okay

### CODE REVIEW

File: .coverage.sh [hunk 1]
```diff
@@ -2,30 +2,21 @@
```
Comment (lippirk) [inline .coverage.sh]: I couldn't get this script to work, so that's why I have rewritten it

File: .travis.yml [hunk 2]
```diff
@@ -1,12 +1,25 @@
...
+    - PACKAGE=rrd-transport
   matrix:
-    - DISTRO="debian-unstable"
+    # We need to pass some Travis environment variables to the container to
+    # enable uploading to coveralls and detection of Travis CI.
+    # Also, set TESTS to false to avoid running them twice.
+    - BASE_REMOTE=git://github.com/xapi-project/xs-opam \
+      TEST=false \
+      POST_INSTALL_HOOK="env TRAVIS=$TRAVIS TRAVIS_JOB_ID=$TRAVIS_JOB_ID bash -ex .coverage.sh"
+    - EXTRA_REMOTES=git://github.com/xapi-project/xs-opam
```
Comment (psafont) [inline .travis.yml]: This should not need to be set, `BASE_REMOTE` is already pointed to xs-opam

File: .travis.yml [hunk 3]
```diff
@@ -1,12 +1,25 @@
...
+matrix:
```
Comment (psafont) [inline .travis.yml]: Is this section needed at all?

**Comment by psafont** [comment]
I think it's worth a shot adding `BISECT_ENABLE: 'YES'` to the env in `.travis.yml` to see if ppx_bisect is working correctly on `.travis.sh` so the tests do not need to be run on the post_install_hook.

Did you need to pass the variables `TRAVIS` and `TRAVIS_JOB_ID` explicitly for the coverage script to pick them up? It seems strange that it's needed.

Good job on getting it to work so fast 👍

**Comment by lippirk** [comment]
> I think it's worth a shot adding `BISECT_ENABLE: 'YES'` to the env in `.travis.yml` to see if ppx_bisect is working correctly on `.travis.sh` so the tests do not need to be run on the post_install_hook.
> 
> Did you need to pass the variables `TRAVIS` and `TRAVIS_JOB_ID` explicitly for the coverage script to pick them up? It seems strange that it's needed.
> 
> Good job on getting it to work so fast +1

I'm about to push your suggestions, will let you know whether it works!

**Comment by lindig** [comment]
This looks good. Please sign off your commits (`git commit -s` -- see the DCO details) and squash this into a single commit.

File: .travis.yml (hunk 3 above)
Comment (lippirk) [reply .travis.yml]: I had a go at removing various parts

the REMOTE variables *and* `TRAVIS=$TRAVIS` do appear to be necessary

**Pull request merged**